### DISCUSSION TITLE
Internal Server Error when filtering on self-referencing m2m relation

### DISCUSSION THREAD

**Discussion by Tapematch** [directus/directus#6202] [Archive]
Hey guys,
I still need help with my issue #5747.
I tried it with the newest RC75, but am still getting the same errors. I created my collections from scratch to rule out any migration errors.

Here a small summary:
1. Create a self-referencing m2m relation
2. Try filtering items based on that relation

This is the filter I'm trying to use on the `article` collection:
```
{
  "blog": {
    "partner_blogs": {
      "related_blog_id": {
        "users": {
          "id": {
            "_eq": "$CURRENT_USER"
          }
        }
      }
    }
  }
}
```

And this the error it generates:
```
error: select "article"."id", "article"."title", "article"."blog" from "article" left join "blog" as "ciakr" on "article"."blog" = "ciakr"."id" left join "blog_blog" as "khjic" on "ciakr"."id" = "khjic"."blog_id" where "directus_users"."id" = $1 order by "article"."id" asc limit $2 - missing FROM-clause entry for table "directus_users"
```

I also attached a new database dump. Maybe someone here can point out my errors. I am grateful for any help. Thanks!

[dump.zip](https://github.com/directus/directus/files/6636660/dump.zip)
_Admin user is admin@example.com:12345678. You can use the token 12345678 for requesting._

**Comment by rijkvanzanten** [comment]
Lets keep this conversation going on #5747 👍🏻
